Beaconchain crashes after sycing

![2020-08-18 11_41_42-](https://user-images.githubusercontent.com/45544213/90553380-24885c80-e149-11ea-84a5-bf046754cc52.png)


After alpha.23 was released, I downloaded and built from source, Ran Beacon chain by clearing db and dev flags. But the process crashes after syncing with the error message in the screenshot. Let me know if you need more info.

Just to be sure, I pulled the latest version again and built it - reran the process without clear-db flag. Started syncing from ~71k again, syncs to head and then crashes -

[2020-08-18 13:55:42]  INFO initial-sync: Processing block 0x08a3a233... 102560/102577 - estimated time remaining 1s blocksPerSecond=10.8 peers=100
[2020-08-18 13:55:42]  INFO blockchain: Synced new block block=0x08a3a233... epoch=3205 finalizedEpoch=2289 finalizedRoot=0xd0ea421f... slot=102560
[2020-08-18 13:55:42]  INFO blockchain: Finished applying state transition attestations=106 attesterSlashings=0 deposits=0 proposerSlashings=0 voluntaryExits=0
[2020-08-18 13:55:42]  INFO blockchain: Synced new block block=0x27ebdc60... epoch=3205 finalizedEpoch=2289 finalizedRoot=0xd0ea421f... slot=102563
[2020-08-18 13:55:42]  INFO blockchain: Finished applying state transition attestations=52 attesterSlashings=0 deposits=0 proposerSlashings=0 voluntaryExits=0
[2020-08-18 13:55:42]  INFO blockchain: Synced new block block=0x148f98f5... epoch=3205 finalizedEpoch=2289 finalizedRoot=0xd0ea421f... slot=102564
[2020-08-18 13:55:42]  INFO blockchain: Finished applying state transition attestations=71 attesterSlashings=0 deposits=0 proposerSlashings=0 voluntaryExits=0
[2020-08-18 13:55:42]  INFO blockchain: Synced new block block=0x9cc4b49d... epoch=3205 finalizedEpoch=2289 finalizedRoot=0xd0ea421f... slot=102565
[2020-08-18 13:55:42]  INFO blockchain: Finished applying state transition attestations=128 attesterSlashings=0 deposits=0 proposerSlashings=0 voluntaryExits=0
[2020-08-18 13:55:42]  INFO blockchain: Synced new block block=0x6d3de7d2... epoch=3205 finalizedEpoch=2289 finalizedRoot=0xd0ea421f... slot=102566
[2020-08-18 13:55:42]  INFO blockchain: Finished applying state transition attestations=68 attesterSlashings=0 deposits=0 proposerSlashings=0 voluntaryExits=0
[2020-08-18 13:55:42]  INFO blockchain: Synced new block block=0x73997a2e... epoch=3205 finalizedEpoch=2289 finalizedRoot=0xd0ea421f... slot=102570
[2020-08-18 13:55:42]  INFO blockchain: Finished applying state transition attestations=113 attesterSlashings=0 deposits=0 proposerSlashings=0 voluntaryExits=0
[2020-08-18 13:55:42]  INFO blockchain: Synced new block block=0x2ec76089... epoch=3205 finalizedEpoch=2289 finalizedRoot=0xd0ea421f... slot=102572
[2020-08-18 13:55:42]  INFO blockchain: Finished applying state transition attestations=121 attesterSlashings=0 deposits=0 proposerSlashings=0 voluntaryExits=0
[2020-08-18 13:55:45]  INFO initial-sync: Synced up to slot 102572
[2020-08-18 13:55:54]  INFO sync: Requesting parent block currentSlot=101510 parentRoot=95d13faf2b20
[2020-08-18 13:55:58]  INFO sync: Requesting parent block currentSlot=97625 parentRoot=7deda53418d2
[2020-08-18 13:56:00]  INFO sync: Requesting parent block currentSlot=101510 parentRoot=95d13faf2b20
[2020-08-18 13:56:00]  INFO blockchain: Synced new block block=0x4588df46... epoch=3205 finalizedEpoch=2289 finalizedRoot=0xd0ea421f... slot=102576
[2020-08-18 13:56:00]  INFO blockchain: Finished applying state transition attestations=82 attesterSlashings=0 deposits=0 proposerSlashings=0 voluntaryExits=0
panic: interface conversion: interface {} is bitfield.Bitlist, not []uint8

goroutine 179 [running]:
github.com/gogo/protobuf/proto.equalAny(0x1861ca0, 0xc05023e240, 0x197, 0x1861ca0, 0xc05023e240, 0x197, 0xc00a1fbd00, 0x1861ca0)
        /home/ddev/go/pkg/mod/github.com/gogo/protobuf@v1.3.1/proto/equal.go:205 +0x161c
github.com/gogo/protobuf/proto.equalStruct(0x184aca0, 0xc05023e240, 0x199, 0x184aca0, 0xc05023e240, 0x199, 0x8)
        /home/ddev/go/pkg/mod/github.com/gogo/protobuf@v1.3.1/proto/equal.go:114 +0x3cc
github.com/gogo/protobuf/proto.equalAny(0x184aca0, 0xc05023e240, 0x199, 0x184aca0, 0xc05023e240, 0x199, 0xc038750900, 0xc00cca4c58)
        /home/ddev/go/pkg/mod/github.com/gogo/protobuf@v1.3.1/proto/equal.go:220 +0xdc1
github.com/gogo/protobuf/proto.equalAny(0x18cb800, 0xc04787a2c0, 0x196, 0x18cb800, 0xc04787a2c0, 0x196, 0xc038750900, 0x5b)
        /home/ddev/go/pkg/mod/github.com/gogo/protobuf@v1.3.1/proto/equal.go:192 +0x935
github.com/gogo/protobuf/proto.equalAny(0x16363a0, 0xc00cfaec98, 0x197, 0x16363a0, 0xc00cfaec98, 0x197, 0xc038750900, 0x16363a0)
        /home/ddev/go/pkg/mod/github.com/gogo/protobuf@v1.3.1/proto/equal.go:212 +0x13a2
github.com/gogo/protobuf/proto.equalStruct(0x18b8900, 0xc00cfaec30, 0x199, 0x18b8900, 0xc00cfaec30, 0x199, 0xc050cb43b8)
        /home/ddev/go/pkg/mod/github.com/gogo/protobuf@v1.3.1/proto/equal.go:114 +0x3cc
github.com/gogo/protobuf/proto.equalAny(0x18b8900, 0xc00cfaec30, 0x199, 0x18b8900, 0xc00cfaec30, 0x199, 0xc00a85ab00, 0x18fc5e0)
        /home/ddev/go/pkg/mod/github.com/gogo/protobuf@v1.3.1/proto/equal.go:220 +0xdc1
github.com/gogo/protobuf/proto.equalStruct(0x186f920, 0xc0134640e0, 0x199, 0x186f920, 0xc0134640e0, 0x199, 0xc050cb4390)
        /home/ddev/go/pkg/mod/github.com/gogo/protobuf@v1.3.1/proto/equal.go:114 +0x3cc
github.com/gogo/protobuf/proto.equalAny(0x186f920, 0xc0134640e0, 0x199, 0x186f920, 0xc0134640e0, 0x199, 0xc00a85a500, 0x18d8260)
        /home/ddev/go/pkg/mod/github.com/gogo/protobuf@v1.3.1/proto/equal.go:220 +0xdc1
github.com/gogo/protobuf/proto.equalStruct(0x182ff00, 0xc033c66140, 0x199, 0x182ff00, 0xc033c66140, 0x199, 0xc030dc4538)
        /home/ddev/go/pkg/mod/github.com/gogo/protobuf@v1.3.1/proto/equal.go:114 +0x3cc
github.com/gogo/protobuf/proto.Equal(0x1d50d80, 0xc033c66140, 0x1d50d80, 0xc033c66140, 0xc00d0bae01)
        /home/ddev/go/pkg/mod/github.com/gogo/protobuf@v1.3.1/proto/equal.go:92 +0x331
github.com/prysmaticlabs/prysm/beacon-chain/sync.(*Service).deleteBlockFromPendingQueue(0xc000d62600, 0x190b0, 0xc033c66140)
        /home/ddev/prysm/beacon-chain/sync/pending_blocks_queue.go:239 +0x10c
github.com/prysmaticlabs/prysm/beacon-chain/sync.(*Service).processPendingBlocks(0xc000d62600, 0x1d5d480, 0xc057f00c00, 0x0, 0x0)
        /home/ddev/prysm/beacon-chain/sync/pending_blocks_queue.go:155 +0x9c7
github.com/prysmaticlabs/prysm/beacon-chain/sync.(*Service).processPendingBlocksQueue.func1()
        /home/ddev/prysm/beacon-chain/sync/pending_blocks_queue.go:32 +0x72
github.com/prysmaticlabs/prysm/shared/runutil.RunEvery.func1(0xc000d76000, 0x28e0040, 0x5b, 0xc000d72420, 0x1d5d3c0, 0xc008f28d00)
        /home/ddev/prysm/shared/runutil/every.go:23 +0xec
created by github.com/prysmaticlabs/prysm/shared/runutil.RunEvery
        /home/ddev/prysm/shared/runutil/every.go:18 +0x136

--- a/beacon-chain/sync/BUILD.bazel
+++ b/beacon-chain/sync/BUILD.bazel
@@ -91,6 +91,7 @@ go_library(
         "@com_github_prometheus_client_golang//prometheus:go_default_library",
         "@com_github_prometheus_client_golang//prometheus/promauto:go_default_library",
         "@com_github_prysmaticlabs_ethereumapis//eth/v1alpha1:go_default_library",
+        "@com_github_prysmaticlabs_go_ssz//:go_default_library",
         "@com_github_sirupsen_logrus//:go_default_library",
         "@io_opencensus_go//trace:go_default_library",
     ],

--- a/beacon-chain/sync/pending_blocks_queue.go
+++ b/beacon-chain/sync/pending_blocks_queue.go
@@ -6,7 +6,8 @@ import (
 	"sort"
 	"sync"
 
-	"github.com/gogo/protobuf/proto"
+	"github.com/prysmaticlabs/go-ssz"
+
 	"github.com/pkg/errors"
 	ethpb "github.com/prysmaticlabs/ethereumapis/eth/v1alpha1"
 	"github.com/prysmaticlabs/prysm/beacon-chain/core/helpers"
@@ -262,7 +263,7 @@ func (s *Service) deleteBlockFromPendingQueue(slot uint64, b *ethpb.SignedBeacon
 	}
 	newBlks := make([]*ethpb.SignedBeaconBlock, 0, len(blks))
 	for _, blk := range blks {
-		if proto.Equal(blk, b) {
+		if ssz.DeepEqual(blk, b) {
 			continue
 		}
 		newBlks = append(newBlks, blk)

